### DISCUSSION TITLE
storing and displaying dates

### DISCUSSION THREAD

**Discussion by johnantoni** [js-fns/date-fns#2779] [Q&A] [answered]
Hi, I'm trying to get my head around this probably super simple but appreciate some help.

When storing a date without a timezone I use:

```
const dt = new Date('2021-10-04'); // new Date() applies the browser's timezone
const DATE_TIME_ISO = 'yyyy-MM-dd\'T\'HH:mm:ss.SSSxxx';
return format(dt, DATE_TIME_ISO);

```
That goes to the backend and gets saved

And then to display just the date I use:

```
const dt = new Date('2021-10-10T04:00:00.000+00.00');
const DATE_FORMAT = 'yyyy-MM-dd';
return format (dt, DATE_FORMAT);

```

Am I doing this right?

Does new Date() take into account daylight saving or should I be using parse() ?

**Comment by tan75** [comment]
@johnantoni  hi,
is this what you are looking for?

```
import { format, parseISO } from 'date-fns';

const dt = new Date('2021-10-04'); // new Date() applies the browser's timezone
const DATE_TIME_ISO = "yyyy-MM-dd'T'HH:mm:ss.SSSxxx";
const result1 = format(dt, DATE_TIME_ISO); //=> 2021-10-04T01:00:00.000+01:00

const DATE_FORMAT = 'yyyy-MM-dd';
const result2 = format(parseISO(result1), DATE_FORMAT); //=> 2021-10-04;
```

**Reply by fturmel** [reply]
@tan75 I don't think this is quite right. If the local time zone has a negative offset, `format` can return the wrong date (`2021-10-03`) because `new Date('2021-10-04')` will create the Date as Oct 4th 2021 midnight UTC (not local).

_EDIT: split my answer in two comments for better context_

**Comment by fturmel** [comment, marked as answer]
I think the following approach would be better, but it may not be suited to all scenarios. There's a greater discussion to be had about what the date represents in the app and whether the full datetime and time zone should be persisted in the database or not. Think a birthdate vs check-in day for a hotel reservation vs log event for example. Context can matter quite a bit.

```js
import { format, parseISO } from "date-fns";

const input = "2021-10-04"; // some input, maybe from a date picker or something? Make sure it's valid.

const dt1 = parseISO(input); // create Date instance, treat input string as local time zone
const dtStringUTC = dt1.toISOString(); // to UTC string for persistence
console.log(dt1.toString()); // Mon Oct 04 2021 00:00:00 GMT-0400 (Eastern Daylight Time)
console.log(dtStringUTC); // 2021-10-04T04:00:00.000Z

const dt2 = new Date(dtStringUTC); // recreate Date instance with UTC string
const dtString = format(dt2, "yyyy-MM-dd");
console.log(dt2.toString()); // Mon Oct 04 2021 00:00:00 GMT-0400 (Eastern Daylight Time)
console.log(dtString); // 2021-10-04
```

Maybe just the original input is enough to be persisted in a DATE field of your database? In that case, you could simply use `parseISO` whenever you need a Date instance to work with. In addition to being simpler, the Date object will always be at midnight local time regardless of the user's time zone.

```js
import { parseISO } from "date-fns";

const input = "2021-10-04";
const dt = parseISO(input);
```

Hope that was helpful!
